src/components/Search/index.jsx

```diff
@@ -235,16 +235,21 @@ export default function Search({ small = false }) {
     }),
   );
 
-  let uniqueTokens = [];
-  let found = {};
-  allTokens &&
-    allTokens.map((token) => {
-      if (!found[token.id]) {
-        found[token.id] = true;
-        uniqueTokens.push(token);
-      }
-      return true;
-    });
+  const uniqueTokens = useMemo(() => {
+    let tokens = [];
+    let found = {};
+
+    allTokens &&
+      allTokens.map((token) => {
+        if (!found[token.id]) {
+          found[token.id] = true;
+          tokens.push(token);
+        }
+        return true;
+      });
```

Comment:
option 1

```suggestion
      allTokens?.forEach((token) => {
        if (!found[token.id]) {
          found[token.id] = true;
          tokens.push(token);
        }
      });
```

suggested option 2 

```suggestion
const tokens = [...new Map(allTokens.map(token =>
  [token.id, token])).values()]
```